syntax = "proto3";

service NumberSortingService {
  rpc SortNumbers (NumberArray) returns (NumberArray) {}
}

message NumberArray {
  repeated int32 numbers = 1;
}
service UrMumJoke {
    rpc TellJoke (theJokeReq) returns (theJokeReply) {}

  }
message theJokeReq{
    string punchline = 1;
    string reason = 2;
}
message theJokeReply{
    string message = 1;
   
}

service EMGClassifierService{
  rpc Classify_Signal(PredictRequest)returns(PredictedSignal){}
}

service PoseHandService{
  rpc PoseHand_manual(PoseRequest)returns(PoseHand){}
}

message PredictedSignal{
  int32 signal=1;
  string base64plot=2;//signal plot as base64
}
message PlaceHolderMsg{

}
message PredictRequest{
  int32 idx_from=1;
  int32 idx_test_pose=2;
  int32 idx_user=3;
}

message PoseRequest{
  int32 pose=1;
}
message PoseHand{
  // string PlaceHolderMsg=1;
}

// "Ur mum is so XXXXXX that she is XXXXXXX"
// ./starter-kit/prepare-sources.sh --proto ./protos/services.proto --flutterDir ./app --pythonDir ./server
// chmod 755 ./starter-kit/prepare-sources.sh;
// chmod 755 ./starter-kit/bundle-python.sh
